components/two-factor-authentication/AuthenticatorSettings.tsx

```diff
@@ -0,0 +1,275 @@
+          type: TOAST_TYPE.SUCCESS,
+          message: <FormattedMessage defaultMessage="Authenticator added" />,
+        });
+
+        if (result.data.addTwoFactorAuthTokenToIndividual.recoveryCodes) {
+          props.onRecoveryCodes(result.data.addTwoFactorAuthTokenToIndividual.recoveryCodes);
+        }
+      } catch (e) {
+        addToast({
+          type: TOAST_TYPE.ERROR,
+          message: i18nGraphqlException(intl, e),
+        });
+      } finally {
+        props.onClose();
+      }
+    },
+    validate(values) {
+      const errors: Record<string, unknown> = {};
+      if (!values.twoFactorAuthenticatorCode) {
+        errors.twoFactorAuthenticatorCode = intl.formatMessage(I18nMessages.REQUIRED);
+      } else {
+        const verified = speakeasy.totp.verify({
+          secret: base32,
+          encoding: 'base32',
+          token: values.twoFactorAuthenticatorCode,
+          window: 2,
+        });
+
+        if (!verified) {
+          errors.twoFactorAuthenticatorCode = intl.formatMessage(I18nMessages.INVALID_TOTP_CODE);
+        }
+      }
+
+      return errors;
+    },
+  });
+
+  return (
+    <StyledModal onClose={props.onClose}>
+      <Box>
+        <Flex gap="20px">
+          <Flex justifyContent="center">
+            <QRCode value={otpAuthUrl} renderAs="svg" size={128} level="L" includeMargin data-cy="qr-code" />
+          </Flex>
+          <TokenBox data-cy="manual-entry-2fa-token">
+            <P>
+              <FormattedMessage
+                id="TwoFactorAuth.Setup.ManualEntry"
+                defaultMessage="Manual entry: {token}"
+                values={{
+                  token: <Code>{base32}</Code>,
+                }}
+              />
+            </P>
+          </TokenBox>
+        </Flex>
+        <form onSubmit={formik.handleSubmit}>
+          <Box>
+            <StyledInputField
+              required
+              mt={2}
+              mb={3}
+              label={<FormattedMessage defaultMessage="Enter your code without any dashes" />}
+              htmlFor="twoFactorAuthenticatorCode"
+              error={formik.touched.twoFactorAuthenticatorCode && formik.errors.twoFactorAuthenticatorCode}
+              {...formik.getFieldProps('twoFactorAuthenticatorCode')}
+            >
+              {inputProps => (
+                <StyledInput
+                  disabled={formik.isSubmitting}
+                  as={StyledInput}
+                  {...inputProps}
+                  width={240}
+                  minHeight={60}
+                  fontSize="20px"
+                  lineHeight="28px"
+                  placeholder="123456"
+                  pattern="[0-9]{6}"
+                  inputMode="numeric"
+                  minLength={6}
+                  maxLength={6}
+                  data-cy="add-two-factor-auth-totp-code-field"
+                />
+              )}
+            </StyledInputField>
+            <Flex mt={4} gap="20px" justifyContent="flex-end">
```

Comment:
Usually, in this type of modal:
```suggestion
            <Flex mt={4} gap="20px" justifyContent="space-between">
```